compiler/rustc_target/src/spec/base/avr_gnu.rs

```diff
@@ -24,11 +23,12 @@ pub fn target(target_cpu: &'static str, mmcu: &'static str) -> Target {
 
             linker: Some("avr-gcc".into()),
             eh_frame_header: false,
-            pre_link_args: TargetOptions::link_args(LinkerFlavor::Gnu(Cc::Yes, Lld::No), &[mmcu]),
-            late_link_args: TargetOptions::link_args(
-                LinkerFlavor::Gnu(Cc::Yes, Lld::No),
-                &["-lgcc"],
-            ),
+            pre_link_args: MaybeLazy::lazy(|| {
+                TargetOptions::link_args(LinkerFlavor::Gnu(Cc::Yes, Lld::No), &["-mmcu=atmega328"])
```

Comment:
The lazy part can be moved to `TargetOptions::link_args` (and similar functions if any) if link args are always lazy.

Reply:
`TargetOptions::link_args` has now returns a `MaybeLazy`.